src/npc.cpp

```diff
@@ -1677,8 +1679,15 @@ void npc::shop_restock()
     if( is_player_ally() ) {
         return;
     }
-    const item_group_id &from = myclass->get_shopkeeper_items();
-    if( from == item_group_id( "EMPTY_GROUP" ) ) {
+
+    std::vector<item_group_id> from;
+    for( const auto &ig : myclass->get_shopkeeper_items() ) {
+        const faction *fac = get_faction();
+        if( !fac || !ig.trust || ig.trust <= fac->trusts_u ) {
```

Comment:
IMO you should allow negative trust values too instead of disabling this check at trust = 0. That way you can have an item group that a faction can trade away to people they hate just a little bit.

Either way, `trust == 0` is much clearer than the implicit conversion in `!trust`.